src/Microsoft.VisualBasic/src/Microsoft/VisualBasic/FileIO/FileSystem.Windows.vb

```diff
@@ -0,0 +1,319 @@
+        Private Shared Sub ShellDelete(ByVal FullPath As String,
+            ByVal ShowUI As UIOptionInternal, ByVal recycle As RecycleOption, ByVal OnUserCancel As UICancelOption, ByVal FileOrDirectory As FileOrDirectory)
+
+            Debug.Assert(FullPath.Length <> 0 AndAlso IO.Path.IsPathRooted(FullPath), "FullPath must be a full path!!!")
+            Debug.Assert(ShowUI <> UIOptionInternal.NoUI, "Why call ShellDelete if ShowUI is NoUI???")
+
+            ' Set fFlags to control the operation details.
+            Dim OperationFlags As ShFileOperationFlags = GetOperationFlags(ShowUI)
+            If (recycle = RecycleOption.SendToRecycleBin) Then
+                OperationFlags = OperationFlags Or ShFileOperationFlags.FOF_ALLOWUNDO
+            End If
+
+            ShellFileOperation(SHFileOperationType.FO_DELETE, OperationFlags, FullPath, Nothing, OnUserCancel, FileOrDirectory)
+        End Sub
+
+        ''' <summary>
+        ''' Calls NativeMethods.SHFileOperation with the given SHFILEOPSTRUCT, notifies the shell of change,
+        ''' and throw exceptions if needed.
+        ''' </summary>
+        ''' <param name="OperationType">Value from SHFileOperationType, specifying Copy / Move / Delete</param>
+        ''' <param name="OperationFlags">Value from ShFileOperationFlags, specifying overwrite, recycle bin, etc...</param>
+        ''' <param name="FullSource">The full path to the source.</param>
+        ''' <param name="FullTarget">The full path to the target. Nothing if this is a Delete operation.</param>
+        ''' <param name="OnUserCancel">Value from UICancelOption, specifying to throw or not when user cancels the operation.</param>
+        Private Shared Sub ShellFileOperation(ByVal OperationType As SHFileOperationType, ByVal OperationFlags As ShFileOperationFlags,
+            ByVal FullSource As String, ByVal FullTarget As String, ByVal OnUserCancel As UICancelOption, ByVal FileOrDirectory As FileOrDirectory)
+
+            Debug.Assert(System.Enum.IsDefined(GetType(SHFileOperationType), OperationType))
+            Debug.Assert(OperationType <> SHFileOperationType.FO_RENAME, "Don't call Shell to rename!!!")
+            Debug.Assert(FullSource.Length <> 0 AndAlso IO.Path.IsPathRooted(FullSource), "Invalid FullSource path!!!")
+            Debug.Assert(OperationType = SHFileOperationType.FO_DELETE OrElse (FullTarget.Length <> 0 AndAlso IO.Path.IsPathRooted(FullTarget)), "Invalid FullTarget path!!!")
+
+
+            ' Get the SHFILEOPSTRUCT
+            Dim OperationInfo As SHFILEOPSTRUCT = GetShellOperationInfo(OperationType, OperationFlags, FullSource, FullTarget)
+
+            Dim Result As Integer
+            Try
+                Result = NativeMethods.SHFileOperation(OperationInfo)
+                ' Notify the shell in case some changes happened.
+                'NativeMethods.SHChangeNotify(SHChangeEventTypes.SHCNE_DISKEVENTS, SHChangeEventParameterFlags.SHCNF_DWORD, IntPtr.Zero, IntPtr.Zero)
```

Comment:
>' [](start = 16, length = 1)

Can this be uncommented?

Reply:
For the previous issue "Avoid integer ...", I don't understand the comment or what needs to be done.
Yes, I just needed to add the functions to NativeMethods.